jmxtrans-output/jmxtrans-output-influxdb/src/main/java/com/googlecode/jmxtrans/model/output/InfluxDbWriter.java

```diff
@@ -178,9 +185,20 @@ public void doWrite(Server server, Query query, Iterable<Result> results) throws
 
 			HashMap<String, Object> filteredValues = newHashMap();
 			Object value = result.getValue();
+
+			ImmutableList<String> typeNamesParam = this.typeNames;
+			// if typeNamesAsTag, we don't concat typeName in values.
+			if (typeNamesAsTags) {
+				typeNamesParam = ImmutableList.<String>of();
+			}
+
+			String key = KeyUtils.getPrefixedKeyString(query, result, typeNames);
 			if (isValidNumber(value)) {
-				String key = KeyUtils.getPrefixedKeyString(query, result, typeNames);
 				filteredValues.put(key, value);
+			}else {
+				if (segregateStringValues) {
+					filteredValues.put(key + "_str", value);
```

Comment:
Up to now, we were just filtering out non numbers (probably a wrong assumption that most backen only support numbers). If InfluxDB actually supports non number values, I'm not sure why we need that `_str` suffix, why not just push any value to influx?

Reply:
I agree. Why do you add this "_str" suffix to the metric name? 
In the current version JMXTrans version non numeric values are not sent to InfluxDB, so the value type should not flicker.

Reply:
My understanding is that @yassineLajmi is interested in collecting those non numeric values for analysis. Since the backend allows collecting non numeric values (at least, that's my understanding), collecting those directly might make sense. I'm not thrilled by the use case, but I can see a need for it.

A better solution would probably to centralize the jmxtrans logs and treat this as a failure of metric collection...